test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -77,14 +80,18 @@ class WasmHttpFilterTest : public Common::Wasm::WasmHttpFilterTestBase<
     setupBase(std::get<0>(GetParam()), code, createContextFn(), root_id, vm_configuration);
   }
   void setupTest(std::string root_id = "", std::string vm_configuration = "") {
-    std::string code =
-        std::get<0>(GetParam()) != "null"
-            ? TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(absl::StrCat(
-                  "{{ test_rundir }}/test/extensions/filters/http/wasm/test_data/",
-                  std::get<1>(GetParam()) == "cpp" ? "test_cpp"
-                                                   : absl::StrCat(vm_configuration, "_rust"),
-                  ".wasm")))
-            : "HttpWasmTestCpp";
+    std::string code;
+    if (std::get<0>(GetParam()) == "null") {
+      code = "HttpWasmTestCpp";
+    } else {
+      if (std::get<1>(GetParam()) == "cpp") {
+        code = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(
+            absl::StrCat("{{ test_rundir }}/", kWasmCppDir, "test_cpp.wasm")));
+      } else {
+        code = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(
+            absl::StrCat("{{ test_rundir }}/", kWasmRustDir, vm_configuration + "_rust.wasm")));
+      }
+    }
```

Comment:
Why did you move all those test files out of their current location? It seems like an unnecessary change that requires extra switches.

Reply:
This is because of the limitation of cargo-raze. We should generate build dependencies per Rust projects. But it will cause more complexity.

Reply:
What's the issue? All those examples have the same dependencies, so `cargo raze` should always generate the same `//cargo`. Also, `cargo raze` is not executed as part of the build process, so it shouldn't really influence the location of checked-in `*.rs` and `*.wasm` files.

Having said that, this is me being pedantic, and if upstream doesn't mind and won't complain about this in the upstream PR, then we can merge this. cc @htuch

Reply:
In this case, processing artifacts from `cargo raze` is pretty complex. Because these should be placed on envoy specific paths. So I'd like to automate these processes. If we do that when Rust test data are scattered and required special library for new Rust test data in the future, it will be difficult to automate them.

Reply:
Is the issue that we're not putting the test `.rs` in the same directory as the test itself? It seems desirable to do this IMHO, otherwise Rust/Wasm is some strange outlier.

Reply:
Also, please don't manually construct paths like you do here, if you want runtime files, use https://github.com/envoyproxy/envoy/blob/466245b50e6d475be841e1c8d5f254dd6ea41272/test/test_common/environment.h#L114

Reply:
Sure. But with desirable approach, we should manually edit cargo-raze generated scripts. IMHO it might have to be tech debt...., because we can't this process automatically.